src/generators/default_map_generator_job.cpp

```diff
@@ -785,17 +762,36 @@ std::string default_map_generator_job::default_generate_map(size_t width, size_t
 	// If the [naming] child is empty, we cannot provide good names.
 	std::map<map_location,std::string>* misc_labels = naming.empty() ? nullptr : labels;
 
-	boost::shared_ptr<name_generator> name_generator;
-	if(naming.has_attribute("name_generator")) {
-		name_generator.reset(new context_free_grammar_generator(naming["name_generator"]));
-		if(!name_generator->is_valid()) {
-			name_generator.reset();
+	name_generator_factory* base_generator_factory = nullptr;
+	name_generator*	base_name_generator            = nullptr;
+	name_generator*	river_name_generator           = nullptr;
+	name_generator*	lake_name_generator            = nullptr;
+	name_generator*	road_name_generator            = nullptr;
+	name_generator*	bridge_name_generator          = nullptr;
+	name_generator*	mountain_name_generator        = nullptr;
+	name_generator*	forest_name_generator          = nullptr;
+	name_generator*	swamp_name_generator           = nullptr;				
+
+	if(misc_labels != nullptr) {
+		base_generator_factory = new name_generator_factory(naming, {"male", "base", "bridge", "road", "river", "forest", "lake", "mountain", "swamp"} );
+		
+		naming.get_old_attribute("base_names", "male_names", "[naming]male_names= is deprecated, use base_names= instead");
+		naming.get_old_attribute("base_name_generator", "male_name_generator", "[naming]male_names= is deprecated, use base_names= instead");
```

Comment:
Incorrect message. You don't really need this one though - since `male_name_generator` is new and not present in any release, it's fine to remove it outright with no deprecation message.